fcrepo-http-api/src/main/java/org/fcrepo/http/api/responses/StreamingBaseHtmlProvider.java

```diff
@@ -236,32 +207,18 @@ private Template getTemplate(final Model rdf, final Node subject,
                                   .findFirst();
 
         if (!template.isPresent()) {
-            LOGGER.trace("Attempting to discover mixin types of node for resource in question: {}", subject);
-            template = rdf.listObjectsOfProperty(createResource(subject.getURI()),
-                                                 createProperty(getRDFNamespaceForJcrNamespace(JCR_NAMESPACE) +
-                                                     "mixinTypes"))
-                          .toList().stream()
-                          .map(x -> x.asLiteral().getLexicalForm())
+            LOGGER.trace("Attempting to discover RDF types of node for resource in question: {}", subject);
+            template = rdf.listObjectsOfProperty(subjectURI, type).toList().stream()
```

Comment:
I don't think you need `toList().stream()` here. You can take the `NodeIterator` result of `listObjectsOfProperty` and use `mapWith` and `filterKeep` to get the same results.